apps/manager_app/manager_app.c

```diff
@@ -78,26 +79,61 @@
  * PRIVATE TYPEDEFS
  *****************************************************************************/
 
-/*****************************************************************************
- * STATIC VARIABLES
- *****************************************************************************/
-
 /*****************************************************************************
  * GLOBAL VARIABLES
  *****************************************************************************/
 
 /*****************************************************************************
  * STATIC FUNCTION PROTOTYPES
  *****************************************************************************/
+/**
+ * @brief Entry point for the manager application of the X1 vault. It is invoked
+ * by the X1 vault firmware, as soon as there is a USB request raised for the
+ * manager app.
+ *
+ * @param usb_evt The USB event which triggered invocation of the manager app
+ */
+void manager_app_main(usb_event_t usb_evt, const void *app_config);
+
+/**
+ * @brief Restricted Entry point for the manager application of the X1 vault. It
+ * is invoked by the X1 vault firmware, as soon as there is a USB request raised
+ * for the manager app.
+ * @note It only allows some functionality such as get device info, device
+ * authentication and firmware update
+ *
+ * @param usb_evt The USB event which triggered invocation of the manager app
+ */
+void manager_app_restricted_main(usb_event_t usb_evt, const void *app_config);
 
 /*****************************************************************************
- * STATIC FUNCTIONS
+ * STATIC VARIABLES
  *****************************************************************************/
+static const cy_app_desc_t manager_desc = {.id = 1,
+                                           .version =
+                                               {
+                                                   .major = 1,
+                                                   .minor = 0,
+                                                   .patch = 0,
+                                               },
+                                           .app = manager_app_main,
+                                           .app_config = NULL};
+
+static const cy_app_desc_t manager_restricted_desc = {
+    .id = 1,
+    .version =
+        {
+            .major = 1,
+            .minor = 0,
+            .patch = 0,
+        },
+    .app = manager_app_restricted_main,
+    .app_config = NULL};
```

Comment:
this goes against a correct design IMO.

Every app should ***have a unique ID*** and ***this is not intuitive***. on every update, you need to update both versions otherwise there will be ambiguity in the version of the manager app.

Reply:
updated